### PR TITLE
Fix CSA_motor_efficiency UnboundLocalError for unsupported pole counts

### PR DESCRIPTION
**Pull request by yuhuavit-hash**
`CSA_motor_efficiency` raises an `UnboundLocalError` (instead of a clear error) when the requested pole count has no standard table: 8-pole motors have no high-efficiency values, and pole counts other than 2/4/6/8 are not in any table.

```python
>>> from fluids.pump import CSA_motor_efficiency
>>> from fluids.constants import hp
>>> CSA_motor_efficiency(100*hp, poles=8, high_efficiency=True)
UnboundLocalError: cannot access local variable 'efficiency'
```

This adds input validation up front, raising a `ValueError` with a clear message for unsupported pole counts (matching the style of `current_ideal`'s phase validation), and documents the supported pole counts in the docstring.

## Changes
- `fluids/pump.py`: validate `poles` before table lookup
- `tests/test_pump.py`: add boundary cases (poles=8 + high_efficiency, poles=3)

## Verification
- `python -m pytest tests/test_pump.py` — 12 passed
- `python -m pytest -m "not slow"` — 1764 passed

### CODE REVIEW

File: fluids/pump.py [hunk 1]
```diff
@@ -375,6 +375,10 @@ def CSA_motor_efficiency(P: float, closed: bool=False, poles: int=2, high_effici
     Several low-efficiency standard high power values were added to allow for
     easy programming; values are the last listed efficiency in the table.
 
+    Only 2, 4, and 6 pole motors have high-efficiency standard values; 8 pole
+    values are available for the minimum efficiency standard only. A
```
Comment (CalebBell) [inline fluids/pump.py]: This sentence makes it sound like the low-standard efficiency table doesn't have 2, 4, or 6 pole motors

**Review by CalebBell — CHANGES_REQUESTED** [review]
Good catch! One bit of feedback.

File: fluids/pump.py (hunk 1 above)
Comment (yuhuavit-hash) [reply fluids/pump.py]: Thanks for the feedback! I've reworded the note in the latest commit to make clear that the minimum efficiency standard also covers 2, 4, and 6 pole motors.

**Review by CalebBell — APPROVED** [review]

**Comment by CalebBell** [comment]
Thank you! This will be in the next release
